### ISSUE TITLE
Calendar View: Select a range to batch create notes

### ISSUE THREAD

**Issue by Acylation** [marcusolsson/obsidian-projects#290]
### What would you like to be added?

The Full-Calendar plugin provides a good exmaple. By simply select the ranges in the table, users are allowed to batch normal notes.

https://user-images.githubusercontent.com/73122375/210030074-872f4e59-5a6c-4918-82a0-e95c9787aa6e.mp4

However, to further improve the experience, if the feature applied, change the behavior of "Drag vertically" from creating tasks from the start day to the end day, like what it does in the video, but apply this to select a vertical range, to create task every specified weekdays. In brief, a range selection like spread sheet is expected.

After triggering the modal, user would batch create notes, or create a recursive task (#69)

### Why is this needed?

High efficiency.

**Comment by andrey-jef** [comment]
Looking at this, I'm curious to learn how the integration of custom view into Projects works? So we can just plug the view of FullCalendar into Projects

**Comment by marcusolsson** [comment]
While I like the idea, this feels like a feature for more advanced users? I'm thinking it may be out of scope for now. Of course, if enough users vote for this, I'll consider it.

**Comment by Acylation** [comment]
> Looking at this, I'm curious to learn how the integration of custom view into Projects works? So we can just plug the view of FullCalendar into Projects

@andrey-jef Here's how the dbfolder plugin realizes it. There are also commits for further support in the repo. https://github.com/RafaelGB/obsidian-db-folder/commit/b58bcd6cac7daeef9509ad5b385ee7c91973a4ce

Seems like much work needs to be done before we can integrate the full-calendar plugin as a view. For now, since the full-calendar plugin also uses YAML metadata as the data source, there's no conflict between the two plugins. We can just include all the required headers in the same note, then use both plugins to manage. 
https://github.com/davish/obsidian-full-calendar/issues/176

For this issue, I intended to introduce "range selection by drag" as a new interaction design, and not that related to the full-calendar plugin actually.

**Comment by Acylation** [comment]
Oh! Sorry for the confusion. Not until I actually tried out the full-calendar plugin did I notice that the "click-and-drag" action will create multi-day events. And in the full-calendar plugin, click-and-drag has nothing to do with the batch operation.

> My initial goal for the Calendar view is to have basic feature parity with calendars that most users are familiar with, for example Google Calendar and Apple Calendar.

Providing users with a common and familiar feature, rather than a new design, will certainly get a better UX. Therefore I would change the title to "click-and-drag interaction". Additionally, for a multi-day event, an additional field is required to describe the start-end or start-lasts relationship, and this needs to be well-designed. The batch operation seems indeed out of the scope of this plugin, it should be discussed in a professional calendar plugin instead.

**Comment by marcusolsson** [comment]
Multi-day events should definitely be supported! Click and drag is a common interaction pattern for this, which I fully support! 👌
